src/nupic/algorithms/Connections.hpp

```diff
@@ -449,41 +441,46 @@ class Connections : public Serializable
    *
    * @retval Number of cells.
    */
-  CellIdx numCells() const;
+  size_t numCells() const { return cells_.size(); }
 
   /**
    * Gets the number of segments.
    *
    * @retval Number of segments.
    */
-  Segment numSegments() const;
+  size_t numSegments() const { 
+	  NTA_ASSERT(segments_.size() >= destroyedSegments_.size()); //TODO remove destroyedSegments_? 
```

Comment:
This assert is not needed.
destroyedSegments_ is very much needed, otherwise segments could never be destroyed.

Reply:
- can I keep the assert?, I guess someone can make a working code-change where #destroyed > #segments_ 

> One computes less than have as much as the other, making it significantly faster. 

can we just rm the SegmentIdx from the list of segments?

Reply:
> can we just rm the SegmentIdx from the list of segments?

No, because then the other segment indexes would change.  We could go through the work of updating these references since they're mostly contained within the connections class, but I'm not sure it would be any faster.